Analysis/src/BaseEventAnalyzer.cpp

```diff
@@ -305,11 +304,68 @@ void BaseEventAnalyzer::ProcessDataSource(const SampleDescriptor& sample, const
                     if(unc_source == UncertaintySource::None) {
                         btag_weights[wp][UncertaintyScale::Up] = static_cast<float>(btag_weight_provider->Get(*event, wp,
                                 UncertaintySource::Eff_b, UncertaintyScale::Up));
-                        btag_weights[wp][UncertaintyScale::Down] = static_cast<float>(btag_weight_provider->Get(*event, wp,
-                                UncertaintySource::Eff_b, UncertaintyScale::Down));
+                        btag_weights[wp][UncertaintyScale::Down] = static_cast<float>(btag_weight_provider->Get(*event,
+                                wp, UncertaintySource::Eff_b, UncertaintyScale::Down));
+                    }
+                }
+
+                std::vector<UncertaintySource> uncs_trigger_weight = { UncertaintySource::EleTriggerUnc,
```

Comment:
btag_weights was a special case. For all the other unc the whole code should be inside `if(unc_source == UncertaintySource::None)`, and not inside each `for`